compiler/rustc_codegen_cranelift/src/abi/pass_mode.rs

```diff
@@ -100,7 +100,7 @@ impl<'tcx> ArgAbiExt<'tcx> for ArgAbi<'tcx, Ty<'tcx>> {
                 }
                 _ => unreachable!("{:?}", self.layout.abi),
             },
-            PassMode::Cast(cast) => cast_target_to_abi_params(cast),
+            PassMode::Cast(ref cast, _) => cast_target_to_abi_params(cast),
```

Comment:
Could you add an assertion that there is no padding so that if you try to compile for a target that uses padding it panics rather than miscompiles while padding is not yet implemented?

Reply:
Just in this one spot? Sure thing.